File: server/bcephal-billing/src/main/java/com/moriset/bcephal/billing/websocket/BillingRequest.java
/**
 * 
 */
package com.moriset.bcephal.billing.websocket;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

/**
 * @author Moriset
 *
 */
@Data
public class BillingRequest {

	public enum BillingRequestType{INVOICE, RUN_OUTCOME}
	
	private BillingRequestType type;
	
	private boolean reprint;
	
	private List<Long> ids;
	
	private boolean fromRepository;
	
	private Long modelId;
	
	public BillingRequest()
	{
		this.ids = new ArrayList<Long>();
	}
	
	public boolean isInvoice() {
		return this.type == BillingRequestType.INVOICE;
	}
	
	public boolean isRunOutcome() {
		return this.type == BillingRequestType.RUN_OUTCOME;
	}
}

File: client/bcephal-client-model/Billing/BillingRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Billing
{
    public class BillingRequest
    {
		public BillingRequestType Type { get; set; }

		public List<long> Ids { get; set; }

		public BillingRequest()
        {
            Ids = new List<long>();
        }
	}
}
